site/app/entities/forum/Thread.php

```diff
@@ -48,12 +75,40 @@ class Thread {
     #[ORM\Column(type: Types::INTEGER)]
     protected int $status;
 
-    #[ORM\Column(type: Types::DATETIMETZ_MUTABLE)]
-    protected DateTime $lock_thread_date;
+    public function getStatus(): int {
+        return $this->status;
+    }
+
+    public function setStatus(int $newStatus): void {
+        $this->status = $newStatus;
+    }
+
+    #[ORM\Column(type: Types::DATETIMETZ_MUTABLE, nullable: true)]
+    protected ?DateTime $lock_thread_date;
+
+    public function getLockDate(): ?DateTime {
+        return $this->lock_thread_date;
+    }
+
+    public function isLocked(): bool {
+        return !is_null($this->lock_thread_date) && $this->lock_thread_date < date("Y-m-d H:i:S");
+    }
 
     #[ORM\Column(type: Types::DATETIMETZ_MUTABLE)]
     protected DateTime $pinned_expiration;
 
+    public function getPinnedExpiration(): DateTime {
+        return $this->pinned_expiration;
+    }
+
+    public function isPinned(): bool {
+        return $this->pinned_expiration > date("Y-m-d H:i:s");
+    }
+
+    public function IsPinnedExpiring(): bool {
+        return $this->pinned_expiration <= date("Y-m-d H:i:s", strtotime("+7 day"));
+    }
```

Comment:
Same comments here about using `DateTime` objects instead of raw `date()` strings.